### PR TITLE
Qualtrics distribution link

### PR DESCRIPTION
**Pull request by SimonLab**
see #53 
This PR needs more environment variable (survey id and distribution id from Qualtrics) see the shared document to get the values

### CODE REVIEW

**Review by Danwhy — APPROVED** [review]
LGTM

File: test/controllers/qualtrics_controller_test.exs [hunk 1]
```diff
@@ -1,8 +1,31 @@
 defmodule ResearchResource.QualtricsControllerTest do
   use ResearchResource.ConnCase
 
-  test "GET /qualtrics/new", %{conn: conn} do
+  setup %{conn: conn} = config do
+    if username = config[:login_as] do
```
Comment (samhstn) [inline test/controllers/qualtrics_controller_test.exs:5]: Nice pattern, I need to start doing this 👍

File: web/controllers/qualtrics/http_client.ex [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+    url = "https://eu.qualtrics.com/API/v3/distributions/#{@qualtrics_distribution_id}/links?surveyId=#{@qualtrics_survey_id}"
+    get_distribution_link(url, user.ttrrid)
+  end
+
+  defp get_distribution_link(url, external_data_reference) do
+    {:ok, res} = HTTPoison.get(url, @qualtrics_headers)
+    {:ok, data} = Poison.Parser.parse(res.body)
+    next_page = data["result"]["nextPage"]
+    elements = data["result"]["elements"]
+    distribution_link = List.first(Enum.filter(elements, fn(elt) -> elt["externalDataReference"] == external_data_reference end))
```
Comment (samhstn) [inline web/controllers/qualtrics/http_client.ex:24]: I may have looked at piping here for readability

**Review by samhstn — APPROVED** [review]

**Comment by samhstn** [comment]
Looks good, merging

File: web/controllers/qualtrics/http_client.ex (hunk 2 above)
Comment (SimonLab) [reply web/controllers/qualtrics/http_client.ex:24]: agree I wasn't sure yet how to make this part of the code a bit nicer, pipeline, another function maybe.

**Pull request merged**